x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/EsqlBinaryComparison.java

```diff
@@ -378,6 +378,9 @@ public Expression singleValueField() {
     private Query translate(TranslatorHandler handler) {
         TypedAttribute attribute = LucenePushdownPredicates.checkIsPushableAttribute(left());
         String name = handler.nameOf(attribute);
+        // Extract the real field name from MultiTypeEsField, and use it in the push down query if it is found
+        String fieldNameFromMultiTypeEsField = LucenePushdownPredicates.extractFieldNameFromMultiTypeEsField(attribute);
+        name = fieldNameFromMultiTypeEsField != null ? fieldNameFromMultiTypeEsField : name;
```

Comment:
Similarly here, I believe the fix in this file is correct but maybe should go into an own PR so we can easily verify to which extent this is tested.